### PR TITLE
   Bug 1894632: Add ES-Proxy Resource

### PR DESCRIPTION
**Pull request by vimalk78**
Bug 1894632: Add ES-Proxy Resource
    
     - Adds proxy reource to logging CR. a manual cherry pick of https://github.com/openshift/cluster-logging-operator/pull/610

/cc @igor-karpukhin @syedriko 
/assign @jcantrill 

### Links
A manual cherry pick of https://github.com/openshift/cluster-logging-operator/pull/610

### CODE REVIEW

File: pkg/k8shandler/defaults.go [hunk 1]
```diff
@@ -8,6 +8,9 @@ var (
 	defaultEsMemory     resource.Quantity = resource.MustParse("16Gi")
 	defaultEsCpuRequest resource.Quantity = resource.MustParse("1")
 
+	defaultEsProxyMemory     resource.Quantity = resource.MustParse("256Mi")
+	defaultEsProxyCpuRequest resource.Quantity = resource.MustParse("100m")
```
Comment (periklis) [inline pkg/k8shandler/defaults.go:12]: Why is this mandatory? The default in 4.5 was `64Mi`. 4.6 and 4.7 only got the bump to `256Mi` for now.

File: pkg/k8shandler/forwarding_test.go [hunk 2]
```diff
@@ -507,22 +507,22 @@ func TestClusterLoggingRequest_generateCollectorConfig(t *testing.T) {
 		{
 			name: "Valid collector config",
 			fields: fields{
-				cluster:           &loggingv1.ClusterLogging{
+				cluster: &loggingv1.ClusterLogging{
```
Comment (periklis) [inline pkg/k8shandler/forwarding_test.go:510]: Is this `go fmt` in play here?

**Review by periklis — CHANGES_REQUESTED** [review]
I suggest to not enable CRD generation in 4.5. It was a tricky thing to do in 4.6 and required annotating the Golang source code to our API carefully. The present state already moves the apiextensions API from v1beta1 to v1 and this should be omitted in 4.5 because of the unsupported CRD generation.

File: pkg/k8shandler/defaults.go (hunk 1 above)
Comment (vimalk78) [reply pkg/k8shandler/defaults.go:12]: this was done based on your PR to change the defaults? are you sure we dont need your PR to be backported?

File: pkg/k8shandler/forwarding_test.go (hunk 2 above)
Comment (vimalk78) [reply pkg/k8shandler/forwarding_test.go:510]: yes

File: deploy/crds/logging.openshift.io_clusterlogforwarders_crd.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (vimalk78) [reply deploy/crds/logging.openshift.io_clusterlogforwarders_crd.yaml:1]: i had changed the CRD manually. copied it from original 4.6 PR though

File: pkg/k8shandler/defaults.go (hunk 1 above)
Comment (vimalk78) [reply pkg/k8shandler/defaults.go]: updated

**Review by jcantrill — COMMENTED** [review]
/approve

File: pkg/k8shandler/logstore.go [hunk 4]
```diff
@@ -192,6 +192,18 @@ func newElasticsearchCR(cluster *logging.ClusterLogging, elasticsearchName strin
...
+				v1.ResourceMemory: defaultEsProxyMemory,
+			},
+			Requests: v1.ResourceList{
+				v1.ResourceMemory: defaultEsProxyMemory,
+				v1.ResourceCPU:    defaultEsProxyCpuRequest,
+			},
+		}
+	}
 
 	if logStoreSpec.NodeCount > 3 {
```
Comment (igor-karpukhin) [inline pkg/k8shandler/logstore.go:208]: The magic number "3"?

Comment (vimalk78) [reply pkg/k8shandler/logstore.go:208]: this is not added in this PR

**Review by periklis — APPROVED** [review]
/lgtm

**Pull request merged**